Arch ARM compile fail: latitude' may be used uninitialized

Thank you for kiwix tools.

In compiling on an Odroid C1 running archarm, using these versions:
$ pkg-config --version
1.5.4
$ ninjapkg-config --version
ninja
$ ninja --version
1.8.2

gcc version 8.2.0 (GCC) 

I get the error below

Best wishes

> $ makepkg
> ==> Making package: kiwix-tools 0.8.0-1 (Sun Dec 30 02:43:16 2018)
> ==> Checking runtime dependencies...
> ==> Checking buildtime dependencies...
> ==> Retrieving sources...
>   -> Found 0.8.0.tar.gz
> ==> Validating source files with sha256sums...
>     0.8.0.tar.gz ... Passed
> ==> Extracting sources...
>   -> Extracting 0.8.0.tar.gz with bsdtar
> ==> Starting build()...
> + exec meson setup --prefix /usr --libdir /usr/lib --libexecdir /usr/lib --bindir /usr/bin --sbindir /usr/bin --includedir /usr/include --datadir /usr/share --mandir /usr/share/man --infodir /usr/share/info --localedir /usr/share/locale --sysconfdir /etc --localstatedir /var --sharedstatedir /var/lib --buildtype release --auto-features enabled --wrap-mode nofallback -D b_lto=false build
> The Meson build system
> Version: 0.49.0
> Source dir: /home/alarm/.cache/yay/kiwix-tools/src/kiwix-tools-0.8.0
> Build dir: /home/alarm/.cache/yay/kiwix-tools/src/kiwix-tools-0.8.0/build
> Build type: native build
> Project name: kiwix-tools
> Project version: 0.8.0
> Appending CXXFLAGS from environment: '-march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16 -O2 -pipe -fstack-protector-strong -fno-plt'
> Appending LDFLAGS from environment: '-Wl,-O1,--sort-common,--as-needed,-z,relro,-z,now'
> Appending CPPFLAGS from environment: '-D_FORTIFY_SOURCE=2'
> Native C++ compiler: c++ (gcc 8.2.0 "c++ (GCC) 8.2.0")
> Build machine cpu family: arm
> Build machine cpu: armv7l
> Dependency threads found: YES 
> Found pkg-config: /usr/bin/pkg-config (1.5.4)
> Dependency kiwix found: YES 3.1.0
> Dependency libmicrohttpd found: YES 0.9.62
> Dependency zlib found: YES 1.2.11
> Program kiwix-compile-resources found: YES (/usr/bin/kiwix-compile-resources)
> DEPRECATION: build_always is deprecated. Combine build_by_default and build_always_stale instead.
> Build targets in project: 5
> Option werror is: True [default: true]
> Found ninja-1.8.2 at /usr/bin/ninja
> ninja: Entering directory `build'
> [2/11] Generating resources with a custom command.
> jquery-ui/jquery-ui.structure.min.css ['jquery_ui', 'jquery_ui_structure_min_css']
> jquery-ui/jquery-ui.min.js ['jquery_ui', 'jquery_ui_min_js']
> jquery-ui/external/jquery/jquery.js ['jquery_ui', 'external', 'jquery', 'jquery_js']
> jquery-ui/images/ui-bg_flat_0_aaaaaa_40x100.png ['jquery_ui', 'images', 'ui_bg_flat_0_aaaaaa_40x100_png']
> jquery-ui/images/ui-bg_flat_75_ffffff_40x100.png ['jquery_ui', 'images', 'ui_bg_flat_75_ffffff_40x100_png']
> jquery-ui/images/ui-icons_222222_256x240.png ['jquery_ui', 'images', 'ui_icons_222222_256x240_png']
> jquery-ui/images/ui-bg_glass_55_fbf9ee_1x400.png ['jquery_ui', 'images', 'ui_bg_glass_55_fbf9ee_1x400_png']
> jquery-ui/images/ui-bg_highlight-soft_75_cccccc_1x100.png ['jquery_ui', 'images', 'ui_bg_highlight_soft_75_cccccc_1x100_png']
> jquery-ui/images/ui-bg_glass_65_ffffff_1x400.png ['jquery_ui', 'images', 'ui_bg_glass_65_ffffff_1x400_png']
> jquery-ui/images/ui-icons_2e83ff_256x240.png ['jquery_ui', 'images', 'ui_icons_2e83ff_256x240_png']
> jquery-ui/images/ui-icons_cd0a0a_256x240.png ['jquery_ui', 'images', 'ui_icons_cd0a0a_256x240_png']
> jquery-ui/images/ui-icons_888888_256x240.png ['jquery_ui', 'images', 'ui_icons_888888_256x240_png']
> jquery-ui/images/ui-bg_glass_75_e6e6e6_1x400.png ['jquery_ui', 'images', 'ui_bg_glass_75_e6e6e6_1x400_png']
> jquery-ui/images/animated-overlay.gif ['jquery_ui', 'images', 'animated_overlay_gif']
> jquery-ui/images/ui-bg_glass_75_dadada_1x400.png ['jquery_ui', 'images', 'ui_bg_glass_75_dadada_1x400_png']
> jquery-ui/images/ui-icons_454545_256x240.png ['jquery_ui', 'images', 'ui_icons_454545_256x240_png']
> jquery-ui/images/ui-bg_glass_95_fef1ec_1x400.png ['jquery_ui', 'images', 'ui_bg_glass_95_fef1ec_1x400_png']
> jquery-ui/jquery-ui.theme.min.css ['jquery_ui', 'jquery_ui_theme_min_css']
> jquery-ui/jquery-ui.min.css ['jquery_ui', 'jquery_ui_min_css']
> home.html.tmpl ['home_html_tmpl']
> include.html.part ['include_html_part']
> taskbar.css ['taskbar_css']
> taskbar.html.part ['taskbar_html_part']
> global_taskbar.html.part ['global_taskbar_html_part']
> opensearchdescription.xml ['opensearchdescription_xml']
> [10/11] Compiling C++ object 'src/server/e19e62e@@kiwix-serve@exe/kiwix-serve.cpp.o'.
> FAILED: src/server/e19e62e@@kiwix-serve@exe/kiwix-serve.cpp.o 
> c++ -Isrc/server/e19e62e@@kiwix-serve@exe -Isrc/server -I../src/server -Istatic/server -I/usr/include/ -I/usr/include/p11-kit-1 -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Werror -std=c++11 -O3 -march=armv7-a -mfloat-abi=hard -mfpu=vfpv3-d16 -O2 -fstack-protector-strong -fno-plt -D_FORTIFY_SOURCE=2 -pthread  -MD -MQ 'src/server/e19e62e@@kiwix-serve@exe/kiwix-serve.cpp.o' -MF 'src/server/e19e62e@@kiwix-serve@exe/kiwix-serve.cpp.o.d' -o 'src/server/e19e62e@@kiwix-serve@exe/kiwix-serve.cpp.o' -c ../src/server/kiwix-serve.cpp
> ../src/server/kiwix-serve.cpp: In function 'MHD_Response* handle_search(RequestContext*)':
> ../src/server/kiwix-serve.cpp:602:29: error: 'longitude' may be used uninitialized in this function [-Werror=maybe-uninitialized]
>          searcher->geo_search(latitude, longitude, distance,
>          ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>                               start, end, isVerbose.load());
>                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ../src/server/kiwix-serve.cpp:602:29: error: 'latitude' may be used uninitialized in this function [-Werror=maybe-uninitialized]
> ../src/server/kiwix-serve.cpp:602:29: error: 'distance' may be used uninitialized in this function [-Werror=maybe-uninitialized]
> cc1plus: all warnings being treated as errors
> ninja: build stopped: subcommand failed.
> ==> ERROR: A failure occurred in build().
>     Aborting...
>

--- a/src/server/kiwix-serve.cpp
+++ b/src/server/kiwix-serve.cpp
@@ -538,9 +538,9 @@ static struct MHD_Response* handle_search(RequestContext* request)
 
   /* Retrive geo search */
   bool has_geo_query = false;
-  float latitude;
-  float longitude;
-  float distance;
+  float latitude = 0;
+  float longitude = 0;
+  float distance = 0;
   try {
     latitude = request->get_argument<float>("latitude");
     longitude = request->get_argument<float>("longitude");

